stmhal/Makefile

```diff
@@ -321,6 +321,16 @@ $(BUILD)/%_$(BOARD).c $(HEADER_BUILD)/%.h $(HEADER_BUILD)/%_af_const.h $(BUILD)/
 $(BUILD)/pins_$(BOARD).o: $(BUILD)/pins_$(BOARD).c
 	$(call compile_c)
 
+GEN_STMCONST_HDR = $(BUILD)/modstm_const.gen.h
+GEN_STMCONST_QSTR = $(BUILD)/modstm_qstr.h
+CMSIS_MCU_LOWER = $(shell $(SED) -n -e 's/\#define[[:space:]]*\(STM32F...x.\)/\1/p' $(wildcard boards/$(BOARD)/stm32f*_hal_conf.h) | tr '[:upper:]' '[:lower:]')
+CMSIS_MCU_HDR = cmsis/devinc/$(CMSIS_MCU_LOWER).h
+
+$(BUILD)/modstm.o: $(GEN_STMCONST_HDR)
+$(BUILD)/%_const_gen.h $(BUILD)/%_qstr.h: $(CMSIS_MCU_HDR) make-stmconst.py
```

Comment:
> Is _gen a typo, and should be .gen? And why not use the constants $(GEN_STMCONST_HDR) and $(GEN_STMCONST_QSTR)?

I was using a pattern rule since both headers are generated from the same process, and didn't want to cause make-stmconst.py to called twice.